File: pigeoncms/Modules/PigeonCms.HelloWord/views/Default.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Web.Caching;
using PigeonCms;
using PigeonCms.Core.Helpers;

public partial class Controls_HelloWord: PigeonCms.BaseModuleControl
{
    protected string LitOutput = "";

    private string yourName = "";
    public string YourName
    {
        get { return GetStringParam("YourName", yourName); }
        set { yourName = value; }
    }

    private string gender = "";
    public string Gender
    {
        get { return GetStringParam("Gender", gender); }
        set { gender = value; }
    }

    private int age = 0;
    public int Age
    {
        get { return GetIntParam("Age", age); }
        set { age = value; }
    }

    protected void Page_Load(object sender, EventArgs e)
    {
        var cache = new CacheManager<string>("PigeonCms.HelloWord");
        if (cache.IsEmpty(this.BaseModule.Id.ToString()))
        {
            LitOutput = "This is an HelloWord module.<br />"
            + "Theese are the params you set for the module; <br />"
            + "your name: " + this.YourName + "<br />"
            + "gender: " + this.Gender + "<br />"
            + "age: " + this.Age.ToString() + "<br />";
            cache.Insert(this.BaseModule.Id.ToString(), LitOutput);
        }
        else
        {
            LitOutput = cache.GetValue(this.BaseModule.Id.ToString());
        }
    }
}

File: pigeoncms/Modules/PigeonCms.HelloWordVb/views/Default.ascx.vb
Imports PigeonCms
Imports PigeonCms.Core.Helpers


Partial Class Controls_HelloWordVb
    Inherits PigeonCms.BaseModuleControl

    Protected LitOutput As String = ""

    Private m_yourName As String = ""
    Public Property YourName() As String
        Get
            Return GetStringParam("YourName", m_yourName)
        End Get
        Set(value As String)
            m_yourName = value
        End Set
    End Property

    Private m_gender As String = ""
    Public Property Gender() As String
        Get
            Return GetStringParam("Gender", m_gender)
        End Get
        Set(value As String)
            m_gender = value
        End Set
    End Property

    Private m_age As Integer = 0
    Public Property Age() As Integer
        Get
            Return GetIntParam("Age", m_age)
        End Get
        Set(value As Integer)
            m_age = value
        End Set
    End Property

    Protected Sub Page_Load(sender As Object, e As EventArgs)
        'example of cache
        Dim cache = New CacheManager(Of String)("PigeonCms.HelloWord")


        If cache.IsEmpty(Me.BaseModule.Id.ToString()) Then
            LitOutput = "This is an HelloWord module.<br />" & "Theese are the params you set for the module; <br />" & "your name: " & Me.YourName & "<br />" & "gender: " & Me.Gender & "<br />" & "age: " & Me.Age.ToString() & "<br />"
            cache.Insert(Me.BaseModule.Id.ToString(), LitOutput)
            LogProvider.Write(Me.BaseModule, "Value created")   'module log example
        Else
            LitOutput = cache.GetValue(Me.BaseModule.Id.ToString())
            LogProvider.Write(Me.BaseModule, "Value from cache")
        End If
    End Sub
End Class
